### DISCUSSION TITLE
How do I install Photon on my main domain?

### DISCUSSION THREAD

**Discussion by xr3less** [Xyphyn/photon#461] [Q&A] [answered]
How do I install Photon on my main domain?
And does installing Lemmy with Ansible prevent me from using Photon?

**Comment by sevonj** [comment, marked as answer]
- Setup a backup lemmy-ui subdomain. l.instance.com seems to be a common choice.
  - You might need to edit some stuff to have certbot request certs for this.
- Run photon on your server (accessible at instance.com:3000 or something)
- Traffic has to be mapped to frontend and lemmy backend, because they both use the same address. The lemmy container does this internally, but now you have to do it in the main nginx conf. You can copy [the map from the internal conf](https://github.com/LemmyNet/lemmy-ansible/blob/5b35eefb66321025ec8e0d58643dfa1eace5e72f/templates/nginx_internal.conf#L32-L47) and change lemmy-ui to photon and lemmy backend to lemmy container.

I recommend getting the first two steps done before touching the third.

> And does installing Lemmy with Ansible prevent me from using Photon?

No. Be careful you don't destroy your changed files when updating lemmy-ansible though.
